### PR TITLE
More misc fixes

### PR DESCRIPTION
**Pull request by rustyrussell**
Including the final one, which fixes a real routefinding issue.

### CODE REVIEW

File: gossipd/routing.c [hunk 1]
```diff
@@ -77,7 +77,6 @@ struct node *new_node(struct routing_state *rstate,
 	n->out = tal_arr(n, struct node_connection *, 0);
 	n->alias = NULL;
 	n->node_announcement = NULL;
-	n->last_timestamp = 0;
```
Comment (cdecker) [inline gossipd/routing.c]: We are returning the channels when asked for them with `getchannels`. If we issue a `getchannels` before receiving the `channel_announcement` and the `channel_update` valgrind will yell at us.

We ourselves will never produce a timestamp that is 0, the above messages refer to duplicate updates that we either sent ourselves and get reflected back from our peers or updates that have travelled two different paths. We can change this to be a signed int64 and use -1 to indicate a missing update if you'd like.

File: gossipd/routing.c [hunk 2]
```diff
@@ -758,7 +755,7 @@ void handle_channel_update(struct routing_state *rstate, const u8 *update, size_
 					    &short_channel_id));
 		tal_free(tmpctx);
 		return;
-	} else if (c->last_timestamp >= timestamp) {
+	} else if (c->channel_update && c->last_timestamp >= timestamp) {
```
Comment (cdecker) [inline gossipd/routing.c]: These checks on the other hand are sensible :+1:

File: .travis.yml [hunk 3]
```diff
@@ -3,12 +3,12 @@ dist: trusty
 sudo: true
 
 env:
-  - NOVALGRIND=1
-  - NOVALGRIND=0
+  - NO_VALGRIND=1
```
Comment (cdecker) [inline .travis.yml]: I sort of liked having the ability to disable valgrind selectively, it allowed us to run part of the tests quicker in one CI build and slower on the other. Merging the two makes one CI run really short and the other even longer. But I agree that if that is the aim we should pick more descriptive variable names.

File: gossipd/routing.c (hunk 1 above)
Comment (rustyrussell) [reply gossipd/routing.c]: Yes, I thought about -1.  Let's do that...

**Comment by rustyrussell** [comment]
> except for the gossipd change which causes a segfault for local connections for which we don't have an update yet and call getchannels

You mean the valgrind warning?

Actually, on review I think we need a grosser change.   The valgrind warning is a good catch: but we should not be initializing feerate_per_kw or delay.  The cleanest way would be to have a pointer to another struct with those and timestamp (which is NULL if we don't know any).  I think this pointer could actually replace the "active" field.

But the minimal change is to change the towire/fromwire functions to detect -1 as a timestamp and know not to marshal/unmarshal the others.  This makes valgrind work on both ends: if you access an uninitialized field you get a warning.

I also stumbled across unused fields and functions...

File: lightningd/gossip_msg.c [hunk 4]
```diff
@@ -58,8 +60,10 @@ void towire_gossip_getchannels_entry(
 	towire_pubkey(pptr, &entry->source);
 	towire_pubkey(pptr, &entry->destination);
 	towire_bool(pptr, entry->active);
-	towire_u32(pptr, entry->fee_per_kw);
-	towire_u32(pptr, entry->delay);
-	towire_u32(pptr, entry->last_update_timestamp);
 	towire_u16(pptr, entry->flags);
+	towire_u64(pptr, entry->last_update_timestamp);
```
Comment (cdecker) [inline lightningd/gossip_msg.c]: Hm, I kind of dislike the idea of pushing this logic down to the serialization layer, but the only alternative I can come up with is passing dummy values like we did before, so I'm ok with doing it this way.

**Pull request merged**